Feature/182 code style

jaapio opened the pull request:
Most changes are applied automatically by executing `phpcbf`. 
Important things to review are:
- Do we agree with all removed tags from docblocks?
- Are there any dangerous typehints added?
 
Fixes #182

allejo commented:
I'll go over this thoroughly at a later time, but do we want to remove the `@since` tag?

jaapio commented:
This is definitely open for discussion. I strictly followed the doctrine coding standards. And didn't look at psr-12.

I agree that we might want to revert the api and since tags

jaapio commented on src/main/php/org/bovigo/vfs/DotDirectory.php:
```diff
@@ -1,35 +1,36 @@
-     * @return  \Iterator
      */
-    public function getIterator(): \Iterator
+    public function getIterator() : Iterator
```
For some reason we decided to use doctrine coding standard. This forces that you are add a use statement all classes, functions and constants. It has been proved solution that resolving functions and constants is slow in php. So I prefer to have them as use statements. because it is hard to force the `\` in front of functions. 

I think it is easier to add all classes in a use statement that distinguish between global and non global classes. I don't know if it does any harm to have them in a use statement.

bizurkur replied:
For the use statements, I don't think it matters if they're there or not. It's all preference. If there's a sniff that fixes it to be one way or the other, it probably doesn't matter.

For the spacing around the `:` in the return type, I definitely think we should try to follow the PSRs and write it as `getFoo(): Foo`. PSRs aren't perfect, but they are the closest thing to having a community standard. If the Doctrine standard is using the [slevomat rule](https://github.com/slevomat/coding-standard#slevomatcodingstandardtypehintsreturntypehintspacing-) for this, we should be able to set `spacesCountBeforeColon` so that it matches the PSR.

jaapio commented on src/main/php/org/bovigo/vfs/content/SeekableFileContent.php:
```diff
@@ -25,34 +31,24 @@ abstract class SeekableFileContent implements FileContent
+    public function read(int $count) : string
     {
-        $data = $this->doRead($this->offset, $count);
+        $data          = $this->doRead($this->offset, $count);
```
Yeah, this is one of the points where I disagree with the doctrine coding standard. We can customize this of course.

jaapio commented on src/main/php/org/bovigo/vfs/vfsStream.php:
```diff
@@ -1,58 +1,80 @@
+use function strlen;
+use function strpos;
+use function substr;
+use function trim;
```
it wasn't a manual fix. it is forced by doctrine
But does it harm to have them? it does harm to have them not. or we have to force all functions to be fully qualified names.

bizurkur replied:
It definitely doesn't harm having them. I think it's all personal preference. Hopefully someone else comments if they have a preference in either direction. 

I know my preference is probably because of the Symfony standard. I've never used the Doctrine standard for any projects before.

jaapio commented on src/main/php/org/bovigo/vfs/vfsStreamWrapper.php:
```diff
@@ -968,70 +987,69 @@ public function dir_opendir(string $path, int $options): bool
+    public function dir_rewinddir() : void
     {
-        return $this->dirIterator->rewind();
+        $this->dirIterator->rewind();
```
problem here is that `rewind` doesn't return. So in that case it should always return true in our case? since `iterator->rewind` will never false

bizurkur replied:
Nice catch.

I would say it seems `vfsStreamContainerIterator::rewind()` should be fixed so it returns a `bool`, then the stream wrapper can stay simple and return whatever value that returns.

Something like this should fix the issue:
```php
// vfsStreamContainerIterator
    public function rewind()
    {
        $success = reset($this->children);

        return $success !== false;
    }
```

jaapio commented:
@bizurkur since this PR is a discussion point I will create follow up PR depending on the outcome of the point we take here. It will be more work to revert things than reapply the standards automatically :-D

bizurkur commented on src/main/php/org/bovigo/vfs/vfsStream.php:
```diff
@@ -63,37 +85,36 @@ class vfsStream
+        return self::SCHEME . '://' . implode(
+            '/',
+            array_map(
+                'rawurlencode',    // ensure singe path parts are correctly urlencoded
```
This could be fixed in a different PR,  since it was already spelled wrong before

bizurkur commented on src/main/php/org/bovigo/vfs/vfsStreamWrapper.php:
```diff
@@ -272,19 +301,21 @@ protected function resolvePath(string $path): string
+     * @param string|null $opened_path full path that was actually opened
      */
-    public function stream_open(string $path, string $mode, int $options, string $opened_path = null): bool
+    public function stream_open(string $path, string $mode, int $options, ?string $opened_path = null) : bool
```
This could be fixed in a different PR,  since it was already wrong before

bizurkur commented on src/main/php/org/bovigo/vfs/vfsStreamWrapper.php:
```diff
@@ -272,19 +301,21 @@ protected function resolvePath(string $path): string
+    public function stream_open(string $path, string $mode, int $options, ?string $opened_path = null) : bool
     {
-        $extended = ((strstr($mode, '+') !== false) ? (true) : (false));
+        $extended = (strstr($mode, '+') !== false ? (true) : (false));
```
This could be fixed in a different PR,  since it was already odd before

bizurkur commented:
I finished looking at the tests.

Not sure about everyone else, but I'd be ok giving this a :+1: if there were an agreement on the removal of `@api` and `@since`. That's the only thing that seems iffy and wouldn't be easy to add back.

There's some whitespace issues some of these fixes caused and some other issues that these fixes highlight, but all of that seems like it could be fixed in a follow up PR.

jaapio commented:
Ok, I reverted the changes in the removed tags, @since and @api are now back in place.

I changed the code style a bit so we do agree with unaligned variable assignments. And removed the space between the closing) an : of the return type.

Anything else before we merge this monster?
